sklearn/datasets/_openml.py

```diff
@@ -377,57 +402,69 @@ def _get_data_info_by_name(name, version, data_home):
 
     # an integer version has been provided
     url = (_SEARCH_NAME + "/data_version/{}").format(name, version)
-    json_data = _get_json_content_from_openml_api(url, None, False,
-                                                  data_home)
-    if json_data is None:
+    try:
+        json_data = _get_json_content_from_openml_api(
+            url, error_message=None, data_home=data_home
+        )
+        retry = False
+    except OpenMLError:
+        retry = True
+    if retry:
         # we can do this in 1 function call if OpenML does not require the
         # specification of the dataset status (i.e., return datasets with a
         # given name / version regardless of active, deactivated, etc. )
         # TODO: feature request OpenML.
         url += "/status/deactivated"
         error_msg = "Dataset {} with version {} not found.".format(name,
                                                                    version)
-        json_data = _get_json_content_from_openml_api(url, error_msg, True,
-                                                      data_home)
+        json_data = _get_json_content_from_openml_api(
```

Comment:
Can this logic be inside the `except OpenMLError` block, removing the need for `retry`?